sympy/combinatorics/fp_groups.py

```diff
@@ -1003,4 +1005,301 @@ def first_in_class(C, Y=[]):
+                    C.P[nu][A_dict_inv[x]] = v**-1
+
+
+def elimination_technique_1(C):
+    rels = list(C.reidemeister_relators)
+    # the shorter relators are examined first so that generators selected for
+    # elimination will have shorter strings as equivalent
+    rels.sort(reverse=True)
+    gens = list(C.schreier_generators)
+    # TODO: redundant generator can also be present as inverse in relator
+    redundant_gens = {}
+    contained_gens = []
+    # examine each relator in relator list for any generator occuring exactly
+    # once
+    next_i = False
+    for i in range(len(rels) -1, -1, -1):
+        rel = rels[i]
+        for gen in redundant_gens:
+            if any([gen.array_form[0][0] == r[0] for r in rel.array_form]):
+                next_i = True
+                break
+        if next_i:
+            next_i = False
+            continue
+        j = len(gens) - 1
+        while j >= 0:
+            gen = gens[j]
+            if rel.generator_exponent_sum(gen) == 1 and gen not in contained_gens:
+                k = rel.exponent_sum_word(gen)
+                gen_index = rel.index(gen**k)
+                bk = rel.subword(gen_index + 1, len(rel))
+                fw = rel.subword(0, gen_index)
+                redundant_gens[gen] = (bk*fw)**(-1*k)
+                contained_gens.extend((bk*fw).contains_generators())
+                del rels[i]; del gens[j]
+                break
+            j -= 1
+    # eliminate the redundant generator from remaing relators
+    for i, gen in product(range(len(rels)), redundant_gens):
+        rels[i] = rels[i].eliminate_word(gen, redundant_gens[gen])
+    rels.sort()
+    C.reidemeister_relators = rels
+    C.schreier_generators = gens
+
+def elimination_technique_2(C):
+    pass
+
+
+def reidemeister_presentation(fp_grp, H):
+    """
+    fp_group: A finitely presented group, an instance of FpGroup
+    H: A subgroup whose presentation is to be found, given as a list
+    of words in generators of `fp_grp`
+
+    Examples
+    ========
+
+    >>> from sympy.combinatorics.free_group import free_group
+    >>> from sympy.combinatorics.fp_groups import FpGroup, reidemeister_presentation
+    >>> F, x, y = free_group("x, y")
+
+    # Example 5.6 from handbook
+    >>> f = FpGroup(F, [x**3, y**5, (x*y)**2])
+    >>> H = [x*y, x**-1*y**-1*x*y*x]
+    >>> reidemeister_presentation(f, H)
+    ([y_1, y_2], [y_1**2, y_2**-3, y_2*y_1*y_2*y_1*y_2*y_1])
+
+    >>> f = FpGroup(F, [x**3, y**3, (x*y)**3])
+    >>> H = [x*y, x*y**-1]
+    >>> reidemeister_presentation(f, H)
+    ([x_0, y_0], [x_0**3, y_0**3, x_0*y_0*x_0*y_0*x_0*y_0])
+
+    # Exercises Q2. Pg 187 from [1]
+    >>> f = FpGroup(F, [x**2*y**2, y**-1*x*y*x**-3])
+    >>> H = [x]
+    >>> reidemeister_presentation(f, H)
+    ([x_1], [x_1**-4, x_1**-8])
+
+    >>> f = FpGroup(F, [x**3*y**-3, (x*y)**3, (x*y**-1)**2])
+    >>> H = [x]
+    >>> reidemeister_presentation(f, H)
+    ([x_0], [<identity>, x_0**-4, x_0**6, x_0**-12])
```

Comment:
This occurs, since of the presence of "removal of dependent relator" in the relator list, I don't exactly understand the definition of "independent relators", I have only have an idea about it (since no official definition, AFAIK).